frontend/app/src/components/home/P2PSwapContentBuilder.svelte

```diff
@@ -0,0 +1,270 @@
+        </span>
+        <form slot="body" class="body">
+            <div class="row">
+                <div class="select-from">
+                    <Legend label={i18nKey("cryptoAccount.transactionHeaders.from")} />
+                    <div class="inner">
+                        <CryptoSelector
+                            filter={(t) => t.balance > 0}
+                            bind:ledger={fromLedger}
+                            on:select={onSelectFromToken} />
+                    </div>
+                </div>
+                <div class="amount">
+                    <TokenInput
+                        ledger={fromLedger}
+                        {minAmount}
+                        maxAmount={fromDetails.balance}
+                        bind:state={tokenInputState}
+                        bind:valid={fromAmountValid}
+                        bind:amount={fromAmount} />
+                </div>
+            </div>
+            <div class="row">
+                <div class="select-to">
+                    <Legend label={i18nKey("cryptoAccount.transactionHeaders.to")} />
+                    <div class="inner">
+                        <CryptoSelector
+                            filter={(t) => t.ledger !== fromLedger}
+                            bind:ledger={toLedger} />
+                    </div>
+                </div>
+                <div class="amount">
+                    <TokenInput
+                        ledger={fromLedger}
+                        bind:valid={toAmountValid}
+                        bind:amount={toAmount} />
+                </div>
+            </div>
+            <div class="duration">
+                <Legend label={i18nKey("p2pSwap.expiryTime")} />
+                <DurationPicker bind:milliseconds={expiresIn} />
+            </div>
+            <div class="message">
+                <Legend label={i18nKey("tokenTransfer.message")} />
+                <TextArea
+                    maxlength={200}
+                    rows={3}
+                    autofocus={false}
+                    placeholder={i18nKey("tokenTransfer.messagePlaceholder")}
+                    bind:value={message} />
+            </div>
+            {#if error !== undefined}
+                <ErrorMessage>{error}</ErrorMessage>
+            {/if}
+        </form>
+        <span slot="footer">
+            <ButtonGroup>
+                <Button small={!$mobileWidth} tiny={$mobileWidth} secondary on:click={cancel}
+                    ><Translatable resourceKey={i18nKey("cancel")} /></Button>
+                <Button
+                    small={!$mobileWidth}
+                    disabled={!valid}
+                    tiny={$mobileWidth}
+                    on:click={onSend}
+                    ><Translatable resourceKey={i18nKey("tokenTransfer.send")} /></Button>
+            </ButtonGroup>
+        </span>
+    </ModalContent>
+</Overlay>
+
+<style lang="scss">
+    :global(.body .row input.amount-val) {
```

Comment:
This looks pretty fishy. And also, since it's global, would probably benefit from some slightly less generic class names to avoid clashes.